include/nbl/builtin/hlsl/limits.hlsl

```diff
@@ -0,0 +1,223 @@
+    // identifies the rounding style used by the type
+    NBL_CONSTEXPR_STATIC_INLINE float_round_style round_style = is_integer ? float_round_style::round_toward_zero : float_round_style::round_to_nearest;
+    // identifies the IEC 559/IEEE 754 floating-point types
+    NBL_CONSTEXPR_STATIC_INLINE bool is_iec559 = !is_integer;
+    // identifies types that represent a finite set of values
+    NBL_CONSTEXPR_STATIC_INLINE bool is_bounded = true;
+
+    // TODO verify this
+    // identifies types that handle overflows with modulo arithmetic
+    NBL_CONSTEXPR_STATIC_INLINE bool is_modulo = is_integer && !is_signed && !is_bool;
+
+    // number of radix digits that can be represented without change
+    NBL_CONSTEXPR_STATIC_INLINE int32_t digits = is_integer ? (is_bool? 1 : 8*size - is_signed) : float_digits;
+    // number of decimal digits that can be represented without change
+    NBL_CONSTEXPR_STATIC_INLINE int32_t digits10 = is_integer ? (is_bool ? 0 : S8 * 2 + S32 + int32_t(!is_signed) * S64) : float_decimal_digits;
+
+    // number of decimal digits necessary to differentiate all values of this type
+    NBL_CONSTEXPR_STATIC_INLINE int32_t max_digits10 = !is_integer ? float_decimal_max_digits : 0;
+
+    // the radix or integer base used by the representation of the given type
+    NBL_CONSTEXPR_STATIC_INLINE int32_t radix = 2;
+
+    // one more than the smallest negative power of the radix that is a valid normalized floating-point value
+    NBL_CONSTEXPR_STATIC_INLINE int32_t min_exponent = !is_integer ? float_min_exponent : 0;
+    // the smallest negative power of ten that is a valid normalized floating-point value
+    NBL_CONSTEXPR_STATIC_INLINE int32_t min_exponent10 = !is_integer ? float_min_decimal_exponent : 0;
+    // 	one more than the largest integer power of the radix that is a valid finite floating-point value
+    NBL_CONSTEXPR_STATIC_INLINE int32_t max_exponent = !is_integer ? float_max_exponent : 0;
+    // the largest integer power of 10 that is a valid finite floating-point value
+    NBL_CONSTEXPR_STATIC_INLINE int32_t max_exponent10 = !is_integer ? float_max_decimal_exponent : 0;
+
+    // identifies types which can cause arithmetic operations to trap
+    NBL_CONSTEXPR_STATIC_INLINE bool traps = false;
+    // identifies floating-point types that detect tinyness before rounding
+    NBL_CONSTEXPR_STATIC_INLINE bool tinyness_before = false;
+    
+    NBL_CONSTEXPR_STATIC_INLINE T min = impl::num_traits<T>::MIN;
+    NBL_CONSTEXPR_STATIC_INLINE T max = impl::num_traits<T>::MAX;
+    NBL_CONSTEXPR_STATIC_INLINE T lowest = is_integer ? min : -max;
+    NBL_CONSTEXPR_STATIC_INLINE T denorm_min = impl::num_traits<T>::DENORM_MIN;
+    NBL_CONSTEXPR_STATIC_INLINE T epsilon = is_integer ? 0 : (T(1) / T(1ull<<(float_digits-1)));
+    NBL_CONSTEXPR_STATIC_INLINE T round_error = is_same<T, bool>::value ? 0 : T(0.5);
+    NBL_CONSTEXPR_STATIC_INLINE T infinity = is_same<T, bool>::value ? 0 : T(1e+300 * 1e+300);
```

Comment:
just use `1.0/0.0` its more sane than relying on some huge constant getting converted to INF (also huge constant is just as susceptible to `--ffast-math` on the CPU)

Reply:
it's uint now